components/esm/creaturestats.hpp

```diff
@@ -36,7 +36,7 @@ namespace ESM
         std::vector<int> mSummonGraveyard;
 
         ESM::TimeStamp mTradeTime;
-        int mGoldPool;
+        unsigned int mGoldPool;
```

Comment:
The mGoldPool in the mwmechanics/creaturestats.hpp is still declared as int.

Reply:
I haven't touched anything yet in apps/openmw yet.This was to fix an issue elsewhere.

Reply:
It should stay int IMO. unsigned int tends to not follow the principle of least astonishment when used in calculation. Besides, if a 31-bit type (signed int) isn't big enough to hold a value, a 32-bit type is unlikely to be a sufficient improvement. There are really no significant benefits to using unsigned here.

Reply:
It was because of this:
https://github.com/OpenMW/openmw/blob/fc38467c1a9d0f76869b8eb3d26b2417f60264c8/apps/essimporter/importacdt.hpp#L45

```
#pragma pack(1)
    struct ACDT
    {
        // Note, not stored at *all*:
        // - Level changes are lost on reload, except for the player (there it's in the NPC record).
        unsigned char mUnknown[12];
        unsigned int mFlags;
        float mBreathMeter; // Seconds left before drowning
        unsigned char mUnknown2[20];
        float mDynamic[3][2];
        unsigned char mUnknown3[16];
        float mAttributes[8][2];
        float mMagicEffects[27]; // Effect attributes: https://wiki.openmw.org/index.php?title=Research:Magic#Effect_attributes
        unsigned char mUnknown4[4];
        unsigned int mGoldPool;
        unsigned char mCountDown; // seen the same value as in ACSC.mCorpseClearCountdown, maybe
                                  // this one is for respawning?
        unsigned char mUnknown5[3];
    };
```

Reply:
I agree. The question is, which one is correct (i.e. compatible with vanilla MW). And if it is unsigned int, should we use the opportunity to switch to a signed 64-bit type internally, meaning for anything that does not interact with legacy file formats. I bet there are some wacky players out there that are not far from breaking the 4 billion limit.

Reply:
Depending on your platform, int has been shifted to mean different things. Perhaps at some point we should consider moving to explicit types: uint64_t ?
https://en.cppreference.com/w/cpp/types/integer

Reply:
For data structures used to load/save definitely. I was under the impression that we are doing that already. But maybe I misremember.
Generally I prefer to use the basic types. In the 64 bit case however using one of the new explicit types is necessary, because Microsoft (long only 32 bit, grrr).
Again, I would caution against a move to unsigned. That has the potential to break things. For example I remember that item counts use out of band values (-1) in some places. Not sure if the same is true for gold values. But it could be. If we make such a change (that I am not in favour of) this would have to be checked carefully.

Reply:
Right, I'll revert this particular change for now and look into it again in another PR.